pandas/_libs/lib.pyx

```diff
@@ -1000,36 +998,37 @@ cdef inline bint c_is_list_like(object obj, bint allow_sets):
     )
 
 
-_TYPE_MAP = {
-    'categorical': 'categorical',
-    'category': 'categorical',
-    'int8': 'integer',
-    'int16': 'integer',
-    'int32': 'integer',
-    'int64': 'integer',
-    'i': 'integer',
-    'uint8': 'integer',
-    'uint16': 'integer',
-    'uint32': 'integer',
-    'uint64': 'integer',
-    'u': 'integer',
-    'float32': 'floating',
-    'float64': 'floating',
-    'f': 'floating',
-    'complex64': 'complex',
-    'complex128': 'complex',
-    'c': 'complex',
-    'string': 'string',
-    'S': 'bytes',
-    'U': 'string',
-    'bool': 'boolean',
-    'b': 'boolean',
-    'datetime64[ns]': 'datetime64',
-    'M': 'datetime64',
-    'timedelta64[ns]': 'timedelta64',
-    'm': 'timedelta64',
-    'interval': 'interval',
-}
+cdef:
+    dict _TYPE_MAP = {
```

Comment:
do we actually get anything from typing this?

one of the pie-in-the-sky goals is to get the cython code close enough to valid python that we can make flake8 work on it; this is going in the other direction

Reply:
> do we actually get anything from typing this?

I looked at the generated C code, and it looks like we don't gain anything from typing this, I assumed we did, but I guess not.